data/src/main/java/com/example/data/WeatherRepository.kt

```diff
@@ -0,0 +1,38 @@
+package com.example.data
+
+import com.example.data.database.WeatherDataBaseSource
+import com.example.data.mappers.transformApiToDB
+import com.example.data.mappers.transformBDToEntity
+import com.example.data.service.WeatherService
+import com.example.data.service.responses.ResponseApi
+import com.example.domain.WeatherRepositoryInterface
+import com.example.domain.entities.ResponseWeatherInfo
+import com.example.domain.utils.ServiceResult
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+class WeatherRepository(
+    private val weatherService: WeatherService,
+    private val weatherDataBaseSource: WeatherDataBaseSource
+) : WeatherRepositoryInterface {
+
+    override fun getWeatherInformationToDB(): Flow<ResponseWeatherInfo> =
+        weatherDataBaseSource.getWeatherInformation().map {
+            it.let { it.transformBDToEntity() }
+        }
+
+    override fun getWeatherInformationToService() {
+        weatherService.getWeatherCityInfoToService().let {
```

Comment:
why do you use "let" and not "with"?

Reply:
in the case the information is null, but could be not necessary